### ISSUE TITLE
osm2pgsql-replication: place table for `replication_status` outside of `public` schema

### ISSUE THREAD

**Issue by JakobMiksch** [openstreetmap/osm2pgsql#1801]
## What version of osm2pgsql are you using?

```
2022-10-27 23:58:30  osm2pgsql version 1.6.0 (1.6.0)
Build: RelWithDebInfo
Compiled using the following library versions:
Libosmium 2.17.3
Proj [API 4] Rel. 7.2.1, January 1st, 2021
Lua 5.3.3
```

## What did you do exactly?

I set up a replication of OSM import using `osm2pgsql-replication init -d <dbname> --osm-file your-extract.pbf`. 
It worked well,  but I could not find an option to place the table for `replication_status` in a custom schema and not in the `public` schema.

## What did you expect to happen?

An option of the `osm2pgsql-replication` to define the schema of the `replication_status` table.

## What did you do to try analyzing the problem?

It looks to me that the `table_name` in the script does not have a suffix for schema and is therefore always placed in the `public` schema , see: https://github.com/openstreetmap/osm2pgsql/blob/master/scripts/osm2pgsql-replication#L537

If this is something that is desired to be included in osm2pgsql, I can try to implement it.

**Comment by joto** [comment]
I think the best option is to have the same command line option for this as osm2pgsql itself: `--middle-schema=SCHEMA`. We already have to `-p, --prefix=PREFIX` which is used in both commands, so this makes sense.

@JakobMiksch Do you want to take a stab at implementing this?

**Comment by JakobMiksch** [comment]
@joto yes, I will approach this issue.

**Comment by lonvia** [comment]
Fixed in #1817.
